notifications/src/main/java/dk/shape/games/notifications/presentation/SubjectNotificationsFragment.kt

```diff
@@ -0,0 +1,138 @@
+package dk.shape.games.notifications.presentation
+
+import android.app.Dialog
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.whenResumed
+import androidx.lifecycle.whenStarted
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import dk.shape.games.notifications.R
+import dk.shape.games.notifications.actions.SubjectNotificationsAction
+import dk.shape.games.notifications.aliases.NotifificationsLoadedListener
+import dk.shape.games.notifications.databinding.FragmentSubjectNotificationsBinding
+import dk.shape.games.notifications.extensions.awareSet
+import dk.shape.games.notifications.extensions.launch
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationSheetViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationSwitcherViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationTypeCollectionViewModel
+import dk.shape.games.notifications.presentation.viewmodels.notifications.SubjectNotificationViewModel
+import dk.shape.games.notifications.usecases.SubjectNotificationUseCases
+import dk.shape.games.toolbox_library.configinjection.ConfigFragmentArgs
+import dk.shape.games.toolbox_library.configinjection.action
+import dk.shape.games.toolbox_library.configinjection.config
+import kotlinx.coroutines.Dispatchers
+
+class SubjectNotificationsFragment : BottomSheetDialogFragment() {
+
+    object Args : ConfigFragmentArgs<SubjectNotificationsAction, SubjectNotificationsConfig>()
+
+    private val config: SubjectNotificationsConfig by config()
+    private val action: SubjectNotificationsAction by action()
+
+    private val interactor: SubjectNotificationUseCases by lazy {
+        SubjectNotificationInteractor(
+            action = action,
+            provideDeviceId = config.provideDeviceId,
+            provideNotifications = config.provideNotifications,
+            notificationsDataSource = config.notificationsDataSource,
+            notificationsEventHandler = config.eventHandler
+        )
+    }
+
+    private val viewSwitcherViewModel: SubjectNotificationSwitcherViewModel by lazy {
+        SubjectNotificationSwitcherViewModel()
+    }
+
+    private val notificationViewModel: SubjectNotificationViewModel by lazy {
```

Comment:
```suggestion
    private val notificationViewModel = SubjectNotificationViewModel( ..)
```
what is the benefit of using lazy in that case, since that view model is gonna be created as soon as onCreateView is called ?

Reply:
we need to wait to access action and config to avoid issues